### PR TITLE
Fix remaining clippy lints from Rust 1.65 update

### PR DESCRIPTION
**Pull request by tmpolaczyk**
Follow-up to #2296

### CODE REVIEW

File: data_structures/src/error.rs [hunk 1]
```diff
@@ -276,7 +276,7 @@ pub enum TransactionError {
     DataRequestWeightLimitExceeded {
         weight: u32,
         max_weight: u32,
-        dr_output: DataRequestOutput,
+        dr_output: Box<DataRequestOutput>,
```
Comment (aesedepece) [inline data_structures/src/error.rs:279]: Very interesting. Do you know why are they pushing for using `Box`es liberally like this?

File: toolkit/src/cli/mod.rs [hunk 2]
```diff
@@ -7,7 +7,7 @@ mod arguments;
 pub(crate) mod commands;
 pub(crate) mod data_requests;
 
-use crate::errors::Error;
+use witnet_toolkit::errors::Error;
```
Comment (aesedepece) [inline toolkit/src/cli/mod.rs:10]: No more `crate::`? :cry:

Comment (tmpolaczyk) [reply toolkit/src/cli/mod.rs:10]: `crate::` imports from `main.rs`, `witnet_toolkit` imports from `lib.rs`

File: data_structures/src/error.rs (hunk 1 above)
Comment (tmpolaczyk) [reply data_structures/src/error.rs:279]: https://rust-lang.github.io/rust-clippy/master/index.html#result_large_err

Basically, without the box, the `TransactionError` enum is too big, and that hurts performance a bit. We could have changed all the uses of `Result<(), TransactionError>` to `Result<(), Box<TransactionError>>` but in this case I preferred to box just this one large field.

**Review by aesedepece — APPROVED** [review]

**Pull request merged**